File: 351002/Khmel/Task310/Services/CommentService.cs
public class CommentService : ICommentService
{
    private readonly IRepository<Comment> _repository;

    public CommentService(IRepository<Comment> repository)
    {
        _repository = repository;
    }

    public CommentResponseTo GetById(long id)
    {
        var comment = _repository.GetById(id);
        if (comment == null)
            throw new KeyNotFoundException($"Comment с id={id} не найден");
        return ToResponse(comment);
    }

    public IEnumerable<CommentResponseTo> GetAll() =>
        _repository.GetAll().Select(ToResponse);

    public CommentResponseTo Create(CommentRequestTo request) =>
        ToResponse(_repository.Create(ToModel(request)));

    public CommentResponseTo Update(CommentRequestTo request) =>
        ToResponse(_repository.Update(ToModel(request)));

    public void Delete(long id)
    {
        if (!_repository.Delete(id))
            throw new KeyNotFoundException($"Comment с id={id} не найден");
    }

    private Comment ToModel(CommentRequestTo dto) => new Comment
    {
        Id = dto.Id,
        StoryId = dto.StoryId,
        Content = dto.Content
    };

    private CommentResponseTo ToResponse(Comment model) => new CommentResponseTo
    {
        Id = model.Id,
        StoryId = model.StoryId,
        Content = model.Content
    };
}

File: 351003/Soika/entities-app/discussion/src/main/kotlin/com/example/entitiesapp/service/CommentService.kt
package com.example.entitiesapp.service

import com.example.entitiesapp.dto.CommentRequestTo
import com.example.entitiesapp.dto.CommentResponseTo
import com.example.entitiesapp.exception.NotFoundException
import com.example.entitiesapp.model.Comment
import com.example.entitiesapp.repository.CommentRepository
import org.springframework.stereotype.Service
import java.util.*
import kotlin.random.Random

@Service
class CommentService(private val repository: CommentRepository) {

    private fun toResponse(entity: Comment) = CommentResponseTo(
        id = entity.id,
        content = entity.content,
        storyId = entity.storyId,
        country = entity.country,
        state = entity.state
    )

    fun getAll(): List<CommentResponseTo> = repository.findAll().map { toResponse(it) }

    fun getById(id: Long): CommentResponseTo =
        repository.findByCommentId(id)
            .map { toResponse(it) }
            .orElseThrow { NotFoundException("Comment $id not found", 40404) }

    fun create(dto: CommentRequestTo): CommentResponseTo {
        val entity = Comment(
            id = dto.id ?: Math.abs(Random.nextLong(1000000)),
            storyId = dto.storyId,
            content = dto.content,
            country = dto.country ?: "Unknown",
            state = dto.state
        )
        val saved = repository.save(entity)
        return toResponse(saved)
    }

    fun update(id: Long, dto: CommentRequestTo): CommentResponseTo {
        val oldEntity = repository.findByCommentId(id)
            .orElseThrow { NotFoundException("Comment $id not found", 40404) }

        repository.delete(oldEntity)

        val newEntity = Comment(
            id = id,
            storyId = dto.storyId,
            content = dto.content,
            country = dto.country ?: oldEntity.country
        )
        return toResponse(repository.save(newEntity))
    }

    fun delete(id: Long) {
        val entity = repository.findByCommentId(id)
            .orElseThrow { NotFoundException("Comment $id not found", 40404) }
        repository.delete(entity)
    }

    fun deleteByStoryId(storyId: Long) {
        repository.deleteAllByStoryId(storyId)
    }
}